File: structural/bridge/bridge.py

# Implementor interface
class Implementor:
    def operation_impl(self):
        pass

# Concrete Implementor A
class ConcreteImplementorA(Implementor):
    def operation_impl(self):
        print("ConcreteImplementorA Operation")

# Concrete Implementor B
class ConcreteImplementorB(Implementor):
    def operation_impl(self):
        print("ConcreteImplementorB Operation")

# Abstraction class
class Abstraction:
    def __init__(self, implementor):
        self._implementor = implementor

    def operation(self):
        self._implementor.operation_impl()

# Refined Abstraction class
class RefinedAbstraction(Abstraction):
    def operation(self):
        print("RefinedAbstraction Operation")
        super().operation()

if __name__ == "__main__":
    abstraction = RefinedAbstraction(ConcreteImplementorA())
    abstraction.operation()

    abstraction = RefinedAbstraction(ConcreteImplementorB())
    abstraction.operation()

File: structural/bridge/Bridge.cs

using System;

// Implementor interface
public interface IImplementor
{
    void OperationImpl();
}

// Concrete Implementor A
public class ConcreteImplementorA : IImplementor
{
    public void OperationImpl()
    {
        Console.WriteLine("ConcreteImplementorA Operation");
    }
}

// Concrete Implementor B
public class ConcreteImplementorB : IImplementor
{
    public void OperationImpl()
    {
        Console.WriteLine("ConcreteImplementorB Operation");
    }
}

// Abstraction class
public class Abstraction
{
    protected IImplementor _implementor;

    public Abstraction(IImplementor implementor)
    {
        _implementor = implementor;
    }

    public virtual void Operation()
    {
        _implementor.OperationImpl();
    }
}

// Refined Abstraction class
public class RefinedAbstraction : Abstraction
{
    public RefinedAbstraction(IImplementor implementor) : base(implementor) { }

    public override void Operation()
    {
        Console.WriteLine("RefinedAbstraction Operation");
        base.Operation();
    }
}

// Client code
class Program
{
    static void Main()
    {
        Abstraction abstraction;

        abstraction = new RefinedAbstraction(new ConcreteImplementorA());
        abstraction.Operation();

        abstraction = new RefinedAbstraction(new ConcreteImplementorB());
        abstraction.Operation();
    }
}
